### PR TITLE
Move templates directory out of gem directory to ~/.metanorma/templates

### PR DESCRIPTION
**Pull request by CAMOBAP795**
- #53

### CODE REVIEW

**Comment by abunashir** [comment]
Thanks, @CAMOBAP795, I like the idea of using a home directory, I think this would also help us to maintain the cache version throughout multiple versions. What do you think about the followings?

* Can we ensure it is safe to write into that directory (maybe check if it's writable or not)?
* What about windows, does home directory needs any additional changes or something?

**Comment by ronaldtse** [comment]
Agree with @abunashir -- @CAMOBAP795 can you help add those checks? Thanks!

**Comment by CAMOBAP795** [comment]
> Can we ensure it is safe to write into that directory (maybe check if it's writable or not)?

@abunashir added

> What about windows, does home directory needs any additional changes or something?

Nope `Dir.home` works perfectly for windows

File: lib/metanorma/cli/generator.rb [hunk 1]
```diff
@@ -18,6 +18,14 @@ def run
           document_path.rmtree
         end
 
+        unless Cli.templates_path_access
+          UI.say(
+            "Sorry, #{dir} directory have no write access for current user\n" \
```
Comment (ronaldtse) [inline lib/metanorma/cli/generator.rb]: `"Sorry, the current user is unable to write to the #{dir} directory.\n`

Comment (CAMOBAP795) [reply lib/metanorma/cli/generator.rb]: Done

**Pull request merged**